### ISSUE TITLE
Adding a Finder tab switches window

### ISSUE THREAD

**Issue by mpercillier** [ianyh/Amethyst#1614]
# Description

Adding a second tab to a Finder window switches windows.

# Applications

Finder and any other application on the same screen or work space

# Steps to reproduce

1. Open a Finder window in the Tall layout. The window occupies the entire screen as it is the only window.
2. Open another application, e.g. TextEdit. The Finder window occupies the left half of the screen, the second occupies the right half of the screen, as expected.
3. Open a new tab in the Finder window. The windows switch places: the Finder window moves to the right, the other application moves to the left.
4. Relaunching Amethyst places the windows back in the correct order.

# Expected behavior

Windows should not switch places automatically (i.e. step 3 listed above should not happen).

# Screenshots

See video.

https://github.com/ianyh/Amethyst/assets/14906564/481d684b-32ab-443f-960a-e38d6a8ec6f3

# Versions:

 - macOS: 14.2
 - Amethyst: 0.20.0 (108)
 
# Debug Info

```
Version: 0.20.0 (108)

OS version: Version 14.2 (Build 23C64)

Screens:
	(0.0, 0.0, 1470.0, 956.0) [(0.0, 0.0, 1470.0, 956.0)]

Configuration:
debug-layout-info: 0
use-canary-build: 0
mod2: (
    option,
    shift,
    control
)
mod1: (
    option,
    shift
)
floating-is-blacklist: 1
window-max-count: 0
follow-space-thrown-windows: 1
screen-padding-bottom: 0
screen-padding-right: 0
window-margin-size: 5
mouse-follows-focus: 0
float-small-windows: 1
smart-window-margins: 1
layouts: (
    tall,
    wide,
    fullscreen,
    column
)
window-resize-step: 5
window-minimum-width: 0
new-windows-to-main: 0
screen-padding-left: 0
enables-layout-hud: 1
window-minimum-height: 0
window-margins: 1
floating: (
)
enables-layout-hud-on-space-change: 1
focus-follows-mouse: 0
ignore-menu-bar: 0
screen-padding-top: 0
restore-layouts-on-launch: 1
```

# Additional context

Behavior also occurs on older machine running macOS 12.7.3

**Comment by cheonhyangzhang** [comment]
Facing the exact same issue. 

It looks like might be related to this previous issue
https://github.com/ianyh/Amethyst/issues/633

**Comment by booxter** [comment]
I have the same behavior, see the recording. In my case, it's Alacritty terminal and I open a new tab there by pressing "Cmd+T". When I close the tab with "Cmd+W", it snaps back to the original grid position.

https://github.com/user-attachments/assets/dc0784db-861d-4903-8dec-3383c92cb7d4

**Comment by llimllib** [comment]
I can repro booxter's behavior in ghostty as well, I'd put a screenshot but it's exactly the same - add a tab and the window swaps

**Comment by ianyh** [comment]
Yeah, I think the heuristic for determining if a tab is a new window or not is just too restrictive.

**Comment by JasperSurmont** [comment]
Same happens with ghostty (terminal) supporting tabs.
